pypdf/_writer.py

```diff
@@ -1065,7 +1065,9 @@ def encrypt(
         user_password: str,
         owner_password: Optional[str] = None,
         use_128bit: bool = True,
-        permissions_flag: UserAccessPermissions = ALL_DOCUMENT_PERMISSIONS,
+        permissions_flag: Union[
+            UserAccessPermissions, Dict[str, bool]
```

Comment:
Should we really complicate this API further? We cannot we just provide a convenience method on `UserAccessPermissions` instead?

Reply:
I'd also prefer not to add the string versions here. I regret adding them in the first place :see_no_evil: